src/EntityTypeInfo.php

```diff
@@ -0,0 +1,161 @@
+   *   Current user.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(AccountInterface $current_user, EntityTypeManagerInterface $entity_type_manager) {
+    $this->currentUser = $current_user;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Adds publish/unpublish operations on profile entities.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity on which to define an operation.
+   *
+   * @return array
+   *   An array of operation definitions.
+   *
+   * @see hook_entity_operation()
+   */
+  public function entityOperation(EntityInterface $entity) {
+    $operations = $this->publishedOperations($entity);
+    $operations += $this->unpublishedOperations($entity);
+
+    return $operations;
+  }
+
+  /**
+   * Adds published operations for the profile entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The profile entity.
+   *
+   * @return array
+   *   An array of operations.
+   */
+  protected function publishedOperations(EntityInterface $entity) {
+    $operations = [];
+
+    if (!$entity->isActive()) {
+      return $operations;
+    }
+
+    // If the profile is published, add an unpublish operation.
+    if (!$entity->isDefault()) {
+      $operations['set_default'] = [
+        'title' => $this->t('Mark as default'),
+        'url' => $entity->toUrl('set-default'),
+        'parameter' => $entity,
+        'weight' => 50,
+      ];
+    }
+
+    // Display an unpublish button.
+    if ($entity->access('unpublish')) {
+      /** @var \Drupal\profile\Entity\ProfileInterface $entity */
+
+      $profile_type_storage = $this->entityTypeManager->getStorage('profile_type');
+      /** @var \Drupal\profile\Entity\ProfileTypeInterface $profile_type */
+      $profile_type = $profile_type_storage->load($entity->bundle());
+
+      $operations['unpublish'] = [
+        'title' => $profile_type->getUnpublishLabel(),
+        'url' => Url::fromRoute('entity.profile.unpublish', [
+          'profile' => $entity->id(),
+        ]),
+        'weight' => 51,
+      ];
+    }
+
+    return $operations;
+  }
+
+  /**
+   * Adds unpublished operations for the profile entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The profile entity.
+   *
+   * @return array
+   *   An array of operations.
+   */
+  protected function unpublishedOperations(EntityInterface $entity) {
+    $operations = [];
+
+    if ($entity->isActive()) {
+      return $operations;
+    }
+
+    if ($entity->access('publish')) {
+      /** @var \Drupal\profile\Entity\ProfileInterface $entity */
```

Comment:
Where does this annotation refer to? Seems that it needs to be removed.

Reply:
Fixed.